std/algorithm/comparison.d

```diff
@@ -1310,6 +1357,14 @@ MaxType!T max(T...)(T args)
     auto e = min(a, b, c);
     assert(is(typeof(e) == double));
     assert(e == 2);
+
+    // supports iterator
+    assert([2,1,4,3].max() == 4);
+
+    // keygetter
+    assert([1,5,2].enumerate.max!"a[1]".index == 1);
+    assert([1,5,2].enumerate.max!"a[1]".value == 5);
+    assert([1,5,2].enumerate.max!"a[1]".array == [1,5]);
```

Comment:
IIUC, "a[1]" is the same as "a.value". Using the latter would be clearer and more consistent. I didn't understand this code until I looked up the docs for enumerate. Ditto below various times.

2nd, the `.array` line is a bit confusing. Clearer might be to say the result of max `== tuple(1, 5)`.
Also, the source array starts with 1,5 - perhaps change the 1 to something unique, just to make the example clearer.